http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/KubernetesServiceDbQueries.scala

```diff
@@ -85,6 +86,24 @@ object KubernetesServiceDbQueries {
                      appQuery.getByIdQuery(appId),
                      labelFilter)
 
+  def hasClusterOperationInProgress(clusterId: KubernetesClusterLeoId)(
+    implicit ec: ExecutionContext
+  ): DBIO[Boolean] =
+    for {
+      clusterOpt <- kubernetesClusterQuery.getMinimalClusterById(clusterId)
+      cluster <- if (clusterOpt.isEmpty)
```

Comment:
Nit: I usually like `match` to unwrap options:
```
cluster <- clusterOpt match {
  case None => DBIO.failed(..)
  case Some(cluster) => cluster
}
```

I guess DBIO doesn't have an analog to cats-effect `F.fromOption(..)`

Reply:
Yeah, I wish it did :(